### PR TITLE
Pbi 848 spatial search config

### PR DESCRIPTION
**Pull request by edlu77**
#848

### CODE REVIEW

File: projects/ccf-eui/src/app/modules/filters/filters.theme.scss [hunk 1]
```diff
@@ -7,13 +7,14 @@
   $background: map-get($theme, background);
   $accent: map-get($theme, accent);
 
-  .outline-button {
+  .outline-button, .spatial-search-button {
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/modules/filters/filters.theme.scss:10]: Same as other comments, should not be part of this pr

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<div class="spatial-search-container">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html:1]: Don't use top level container div. Place styles directly in the `:host` selector instead

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  <div class="row1">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html:2]: Improve class name

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      <mat-radio-button class="radio" value="male">Male</mat-radio-button>
+      <mat-radio-button class="radio" value="female">Female</mat-radio-button>
+    </mat-radio-group>
+  </div>
+  
+  <div class="row3">
+    <div class="label">Select an organ</div>
+    <mat-form-field class="spatial-search-form-field" appearance="fill">
+      <mat-label>Organs</mat-label>
+      <mat-select panelClass="organ-select" (selectionChange)="updateOrgan($event.value)">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html:26]: Does this set an initial value if provided?

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    <div class="label">Donor Sex: </div>
+    <mat-radio-group class="radio-group" [(ngModel)]="sex" (change)="updateSex($event.value)">
+      <mat-radio-button class="radio" value="male">Male</mat-radio-button>
+      <mat-radio-button class="radio" value="female">Female</mat-radio-button>
+    </mat-radio-group>
+  </div>
+  
+  <div class="row3">
+    <div class="label">Select an organ</div>
+    <mat-form-field class="spatial-search-form-field" appearance="fill">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html:24]: No reason to prefix with `spatial-search`

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<div class="spatial-search-container">
+  <div class="row1">
+    <div class="col1">
+      <div class="title">Configure Spatial Search</div>
+      <button class="info button">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html:5]: mat-icon-button?

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      <mat-label>Organs</mat-label>
+      <mat-select panelClass="organ-select" (selectionChange)="updateOrgan($event.value)">
+        <mat-option *ngFor="let organ of filteredOrgans" [value]="organ">
+          {{organ.name}}
+        </mat-option>
+      </mat-select>
+    </mat-form-field>
+  </div>
+  
+  <button class="continue-button" mat-button (click)="buttonClicked()">Continue</button>
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.html:34]: Should not be able to click  continue without selecting an organ first

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  @Output() readonly organChange = new EventEmitter<OrganInfo>();
+
+  @Output() readonly itemSelected = new EventEmitter<{ sex: Sex; organ: OrganInfo | undefined }>();
+
+  filteredOrgans: OrganInfo[];
+
+  sex: Sex = 'male';
+
+  constructor(public dialogRef: MatDialogRef<SpatialSearchConfigComponent>) {
+    this.filterOrgans();
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:39]: This is done to early. Inputs have not been bound at this time.

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+  @Output() readonly sexChange = new EventEmitter<Sex>();
+
+  @Output() readonly organChange = new EventEmitter<OrganInfo>();
+
+  @Output() readonly itemSelected = new EventEmitter<{ sex: Sex; organ: OrganInfo | undefined }>();
+
+  filteredOrgans: OrganInfo[];
+
+  sex: Sex = 'male';
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:36]: Needs to be correctly set if an selectedOrgan is passed in

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  updateOrgan(organ: OrganInfo): void {
+    this.selectedOrgan = organ;
+    this.organChange.emit(this.selectedOrgan);
+  }
+
+  buttonClicked(): void {
+    this.itemSelected.emit({ sex: this.sex, organ: this.selectedOrgan });
+  }
+
+  close(): void {
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:57]: Should be part of the behavior component

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+  @Input() allOrgans: OrganInfo[] = ALL_POSSIBLE_ORGANS;
+
+  @Input() selectedOrgan?: OrganInfo;
+
+  @Output() readonly sexChange = new EventEmitter<Sex>();
+
+  @Output() readonly organChange = new EventEmitter<OrganInfo>();
+
+  @Output() readonly itemSelected = new EventEmitter<{ sex: Sex; organ: OrganInfo | undefined }>();
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:32]: `organ` should never be undefined when this is emitted

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+@Component({
+  selector: 'ccf-spatial-search-config',
+  templateUrl: './spatial-search-config.component.html',
+  styleUrls: ['./spatial-search-config.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SpatialSearchConfigComponent {
+  @HostBinding('class') readonly className = 'ccf-spatial-search-config';
+
+  @Input() allOrgans: OrganInfo[] = ALL_POSSIBLE_ORGANS;
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:24]: Should be named `organs`

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+  @Input() selectedOrgan?: OrganInfo;
+
+  @Output() readonly sexChange = new EventEmitter<Sex>();
+
+  @Output() readonly organChange = new EventEmitter<OrganInfo>();
+
+  @Output() readonly itemSelected = new EventEmitter<{ sex: Sex; organ: OrganInfo | undefined }>();
+
+  filteredOrgans: OrganInfo[];
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:34]: Needs to be updated when `organs` inputs changes. Use ngOnChanges to detect and update

File: projects/ccf-eui/src/themes/_dark.scss [hunk 14]
```diff
@@ -70,7 +71,9 @@ $background: map-merge(mat.$dark-theme-background-palette, (
   mat-slider-primary: #ececec,
   info-button-hover: map-get($primary-colors, 850),
   checkbox-color: #636363,
-  button-background-hover: #2f3346
+  button-background-hover: #2f3346,
+  icon-background-hover: #2f3346,
```
Comment (axdanbol) [inline projects/ccf-eui/src/themes/_dark.scss:75]: Need to verify these settings with @libby-maier.
Also needs improved naming!

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+  close(): void {
+    document.getElementsByClassName('modal-animated')[0]?.classList.add('modal-animate-fade-out');
+    setTimeout(() => {
+      this.dialogRef.close();
+    }, 250);
+  }
+
+  filterOrgans(): void {
+    this.filteredOrgans = this.allOrgans.filter(organ => organ.hasSex ?? organ.sex === this.sex);
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts:65]: Incorrect filtering logic. Organs without a sex should always be included. Correct logic: `!organ.hasSex || organ.sex === this.sex`

File: projects/ccf-eui/src/themes/_dark.scss (hunk 14 above)
Comment (libby-maier) [reply projects/ccf-eui/src/themes/_dark.scss:75]: Let me know if you want to jump on a quick call @axdanbol & @edlu77!

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts (hunk 15 above)
Comment (edlu77) [reply projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts]: hasSex = true actually means there is no sex, it might be backwards 🤷‍♂️

Comment (bherr2) [reply projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts]: It means there is both a male and female version

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts (hunk 12 above)
Comment (bherr2) [reply projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts]: Actually, you shouldn't be pulling this from ccf-shared. You should get it from the SceneState: <https://github.com/hubmapconsortium/ccf-ui/blob/fe6d65c8cbe3c81e1e99c5986c8ea418da183275/projects/ccf-eui/src/app/core/store/scene/scene.state.ts#L51>.

File: projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts (hunk 15 above)
Comment (axdanbol) [reply projects/ccf-eui/src/app/shared/components/spatial-search-config/spatial-search-config.component.ts]: Ah, confusing naming... Then your logic is definitely correct

**Comment by edlu77** [comment]
Superceded by #849
